### ISSUE TITLE
Inheriting from widgets gives TypeError on v0.16.0

### ISSUE THREAD

**Issue by kaHaleMaKai** [qtile/qtile#1830]
# Issue description

When subclassing the included widgets, a TypeError is raised.

In my config from 0.15.1, I have subclassed a few widgets. When updating today to 0.16.0, I got the below error.

# Qtile version

v0.16.0 (commit: 340a8386b28e4dd0e1055f2ebd48c1c07d7eae39)

# Stack traces
```python
2020-07-21 13:44:55,780 ERROR libqtile confreader.py:load():L103 Could not import config file '/home/lars/.config/qtile/config.py'
Traceback (most recent call last):
  File "/home/lars/local/qtile/libqtile/confreader.py", line 99, in load
    config = __import__(name)  # noqa: F811
  File "/home/lars/.config/qtile/config.py", line 20, in <module>
    from bar import get_bar
  File "/home/lars/.config/qtile/bar.py", line 21, in <module>
    class Pomodoro(widget.Pomodoro):
TypeError: function() argument 'code' must be code, not str
2020-07-21 13:44:55,780 ERROR libqtile qtile.py:make_qtile():L129 Error while reading config file (Traceback (most recent call last):
  File "/home/lars/local/qtile/libqtile/confreader.py", line 99, in load
    config = __import__(name)  # noqa: F811
  File "/home/lars/.config/qtile/config.py", line 20, in <module>
    from bar import get_bar
  File "/home/lars/.config/qtile/bar.py", line 21, in <module>
    class Pomodoro(widget.Pomodoro):
TypeError: function() argument 'code' must be code, not str
)
Traceback (most recent call last):
  File "/home/lars/local/qtile/libqtile/confreader.py", line 99, in load
    config = __import__(name)  # noqa: F811
  File "/home/lars/.config/qtile/config.py", line 20, in <module>
    from bar import get_bar
  File "/home/lars/.config/qtile/bar.py", line 21, in <module>
    class Pomodoro(widget.Pomodoro):
TypeError: function() argument 'code' must be code, not str

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lars/local/qtile/libqtile/scripts/qtile.py", line 127, in make_qtile
    config = confreader.Config.from_file(options.configfile, kore=kore)
  File "/home/lars/local/qtile/libqtile/confreader.py", line 85, in from_file
    cnf.load()
  File "/home/lars/local/qtile/libqtile/confreader.py", line 105, in load
    raise ConfigError(tb)
libqtile.confreader.ConfigError: Traceback (most recent call last):
  File "/home/lars/local/qtile/libqtile/confreader.py", line 99, in load
    config = __import__(name)  # noqa: F811
  File "/home/lars/.config/qtile/config.py", line 20, in <module>
    from bar import get_bar
  File "/home/lars/.config/qtile/bar.py", line 21, in <module>
    class Pomodoro(widget.Pomodoro):
TypeError: function() argument 'code' must be code, not str
```

# Configuration
Minimal example: I have used the default config and simply added
```python
from libqtile import widget

class Pomodoro(widget.Pomodoro):
    pass
```

**Comment by tych0** [comment]
Ah, yes, I had forgotten about this problem. @ramnes we had a few other reports of this as well.

**Comment by ramnes** [comment]
Hey @kaHaleMaKai. Sorry for the config breakage, looks like we forgot to add this to the changelog.

Imports from `libqtile.widget` are now made through a function proxy to avoid the side effects of importing all widgets at once, which means that you now have to import directly from your widget module if you want to subclass (`from libqtile.widget.pomodoro import Pomodoro`).

**Comment by tych0** [comment]
Is there a way we can fix this?

**Comment by ramnes** [comment]
Dropping Python <3.7 support and using PEP 562 mechanisms is the only alternative to the current implementation if we want to keep the vanity imports without side effects.

**Comment by kaHaleMaKai** [comment]
Thanks for the explanation, everyone. And no worries @ramnes I couldn't be more thankful for this great wm and all the effort you pour into it.

Besides, importing the widget explicitely is to be appriciated.

**Comment by kaHaleMaKai** [comment]
Just want to confirm, that the above changes have fixed the issues for me. Thank you for that instant feedback! Already happy for the upgrade :-)

**Comment by tych0** [comment]
I'm not sure we can drop support for everything < 3.7 right now, as stuff like 3.5 is still in use in Ubuntu 18.04, and I presume other distros as well :(

**Comment by ramnes** [comment]
I think we said at some point that we wanted to only support the last 3 releases of Python? If so, we would drop < 3.7 when 3.9 is out, which is planned for October.

**Comment by ramnes** [comment]
By the way, we already have`python_requires >= 3.6` in `setup.cfg`, so we already dropped Python 3.5 support. Let's drop 3.6 now that 3.9 is out, what do you think @tych0?

Paging @tcld

**Comment by ramnes** [comment]
We dropped Python 3.5 and 3.6 in #1938, so let's fix this with #2048. :)

**Comment by krisfris** [comment]
Just set up Qtile for the first time and the default configuration was still using the old pattern `from libqtile import widget`. Wasn't hard to fix. I can share my configuration if that's helpful as I haven't made other changes yet.
